_P101_TMP117.ino

```diff
@@ -0,0 +1,269 @@
+        optionValues[0] = (0x00);
+        options[0] = F("0 - ConvTime C15mS5");
+        optionValues[1] = (0x01);
+        options[1] = F("1 - ConvTime C125mS");
+        optionValues[2] = (0x02);
+        options[2] = F("2 - ConvTime C250mS");
+        optionValues[3] = (0x03);
+        options[3] = F("3 - ConvTime C500mS");
+        optionValues[4] = (0x04);
+        options[4] = F("4 - ConvTime C1S");
+        optionValues[5] = (0x05);
+        options[5] = F("5 - ConvTime C4S");
+        optionValues[6] = (0x06);
+        options[6] = F("6 - ConvTime C8S");
+        optionValues[7] = (0x07);
+        options[7] = F("7 - ConvTime C16S");
+
+        string += F("<TR><TD>ConvTime:<TD><select name='plugin_101_ConvTime'>");
+        for (byte x = 0; x < TMP117_ConvTime_OPTION; x++)
+        {
+          string += F("<option value='");
+          string += optionValues[x];
+          string += "'";
+          if (choice == optionValues[x])
+            string += F(" selected");
+          string += ">";
+          string += options[x];
+          string += F("</option>");
+        }
+        string += F("</select>");
+
+        byte choice1 = Settings.TaskDevicePluginConfig[event->TaskIndex][1];
+        String options1[TMP117_Averaging_OPTION];
+        uint optionValues1[TMP117_Averaging_OPTION];
+        optionValues1[0] = (0x00);
+        options1[0] = F("0 - Averaging (0)");
+        optionValues1[1] = (0x01);
+        options1[1] = F("1 - Averaging (8)");
+        optionValues1[2] = (0x02);
+        options1[2] = F("2 - Averaging (32)");
+        optionValues1[3] = (0x03);
+        options1[3] = F("3 - Averaging (64)");
+        
+        string += F("<TR><TD>Averaging:<TD><select name='plugin_101_Averaging'>");
+        for (byte x = 0; x < TMP117_Averaging_OPTION; x++)
+        {
+          string += F("<option value='");
+          string += optionValues1[x];
+          string += "'";
+          if (choice1 == optionValues1[x])
+            string += F(" selected");
+          string += ">";
+          string += options1[x];
+          string += F("</option>");
+        }
+        string += F("</select>");
+        string += F("<TR><TD>Sensor Adjustment [&#176;C]:<TD><input type='text' title='calibrate sensor to target temperature' name='");
+        string += F("plugin_101_Adjust' value='");
+        string += Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2];
+        string += F("'>");
+        string += F("<TR><TD>Offset [&#176;C]:<TD><input type='text' title='set temperature offset like ' name='");
+        string += F("plugin_101_Offset' value='");
+        string += Settings.TaskDevicePluginConfigFloat[event->TaskIndex][3];
+        string += F("'>");
+        
+       // LoadTaskSettings(event->TaskIndex); // we need to restore our original taskvalues!
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+        String plugin1 = WebServer.arg("plugin_101_ConvTime");

```

Comment:
Please use the F() macro here, as this way of using regular C-strings is using a lot of memory.